.github/workflows/tag_release.yml

```diff
@@ -1,114 +1,28 @@
+  goreleaser:
     runs-on: ubuntu-latest
     steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Unshallow
+      run: git fetch --prune --unshallow
     - name: Install Go
       uses: actions/setup-go@v1
       with:
         go-version: '1.14.x'
-    - name: Checkout code
-      uses: actions/checkout@v1
-    - name: Get version
-      run: |
-        echo ::set-env name=VERSION::${GITHUB_REF/refs\/tags\//}
-    - name: Build release binaries
-      env:
-        PAIR: ${{ matrix.pair }}
-      shell: bash
-      run: |
-        GOOS=$(echo $PAIR | cut -d'-' -f1)
-        GOARCH=$(echo $PAIR | cut -d'-' -f2)
-        if [[ $GOOS =~ "windows" ]]
-        then
-          env VERSION=$VERSION GOOS=$GOOS GOARCH=$GOARCH make release-exe
-        else
-          env VERSION=$VERSION GOOS=$GOOS GOARCH=$GOARCH make release
-        fi
-    - name: Get upload URL
-      uses: actions/download-artifact@v1
-      with:
-        name: release
-    - name: Set upload URL env var
+    - name: Set GOVERSION
       run: |
-        value=`cat release/upload_url.txt`
-        echo ::set-env name=UPLOAD_URL::$value
-    - name: Upload release asset
-      uses: actions/upload-release-asset@v1.0.1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        upload_url: "${{ env.UPLOAD_URL }}"
-        asset_path: "./fastly_${{ env.VERSION }}_${{ matrix.pair }}.tar.gz"
-        asset_name: "fastly_${{ env.VERSION }}_${{ matrix.pair }}.tar.gz"
-        asset_content_type: application/gzip
-    - name: Persist binary
-      uses: actions/upload-artifact@v1
-      with:
-        name: bin
-        path: ./fastly_${{ env.VERSION }}_${{ matrix.pair }}.tar.gz
-  checksums:
-    needs: upload
-    runs-on: ubuntu-latest
-    steps:
-    - name: Get binaries
-      uses: actions/download-artifact@v1
-      with:
-        name: bin
-    - name: Set version env var
-      run: |
-        echo ::set-env name=VERSION::${GITHUB_REF/refs\/tags\//}
-    - name: Get upload URL
-      uses: actions/download-artifact@v1
-      with:
-        name: release
-    - name: Set upload URL env var
-      run: |
-        value=`cat release/upload_url.txt`
-        echo ::set-env name=UPLOAD_URL::$value
-    - name: Generate checksums
-      run: |
-        cd bin
-        for filename in *; do
-          echo -e "$(sha256sum $filename)" >> fastly_${{ env.VERSION }}_SHA256SUMS
-        done
-    - name: Upload checksums asset
-      uses: actions/upload-release-asset@v1.0.1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        upload_url: ${{ env.UPLOAD_URL }}
-        asset_path: ./bin/fastly_${{ env.VERSION }}_SHA256SUMS
-        asset_name: fastly_${{ env.VERSION }}_SHA256SUMS
-        asset_content_type: text/plain
+        echo "::set-env name=GOVERSION::$(go version)"
+    - name: Run GoReleaser
+      uses: goreleaser/goreleaser-action@v1
+      with:
+        version: latest
```

Comment:
any concern about goreleaser being unstable? I guess time will tell.

Reply:
This is the version of our repository that goreleaser will use as the release version, not the version of goreleaser itself. Sadly, the action doesn't allow you to pin to a version, which was flagged during our security audit.  

`latest` here means use the latest semver-like tag. However we could be more robust here and pass in the tag that triggered the build.

Reply:
Have now updated to use the git tag which triggered the build.